ansible_runner/streaming.py

```diff
@@ -60,12 +62,18 @@ def run(self):
         return self.status, self.rc
 
 
-class Worker(object):
-    def __init__(self, _input=None, _output=None, **kwargs):
+class Worker:
+    def __init__(self, _input=None, _output=None, keepalive_seconds: int = 0, **kwargs):
         if _input is None:
             _input = sys.stdin.buffer
         if _output is None:
             _output = sys.stdout.buffer
+
+        self._keepalive_interval_sec = keepalive_seconds
+        self._keepalive_thread: Thread | None = None
```

Comment:
I might show my own ignorance with this question - but could you explain the python syntax of this line? I try to reproduce it, and I get something like this

```
----> 1 _keepalive_thread: Thread | None = None

TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

Where there is a highlight around "Thread | None"

Reply:
`from typing import Optional, Union`
https://peps.python.org/pep-0484/#union-types says to use `Optional[T]` as shorthand for `Union[T, None]`, but apparently Python 3.10 added support for `<T> | None`. If we need to stay compatible with 3.9, we should probably stick to `Optional[T]`, so here it should be `self._keepalive_thread: Optional[Thread]`

Reply:
It's a Python 3.10+ typing shorthand feature for `Union` (eg, `X | Y | None` is equivalent to `typing.Optional[typing.Union[X, Y]]`), but since I added the `annotations` future, it's allowed and supported back to Python 3.7 from the runtime's perspective, and at least `mypy` and pycharm validate it properly. That future causes the parser to store all type annotations as strings instead of trying to AST them, which allows things like forward references, newer typing syntax in older Pythons, and cheaper runtime overhead.

Reply:
Checks out

```python
>>> from __future__ import annotations
>>> _keepalive_thread: Thread | None = None
```

This works, and the result is that `_keepalive_thread` is `None`, but known to be either a `NoneType` or a `Thread`.